main.go

```diff
@@ -1,29 +1,110 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/elb"
-	etcdClient "github.com/coreos/etcd/client"
-	etcdContext "golang.org/x/net/context"
+	"github.com/jawher/mow.cli"
+	"io"
 	"io/ioutil"
 	"log"
+	"net"
 	"net/http"
 	"os"
 	"strings"
 	"time"
 )
 
-const (
-	konsDNSEndPoint = "https://dns-api.in.ft.com/v2"
-)
+var httpClient = http.Client{
+	Transport: &http.Transport{
+		MaxIdleConnsPerHost: 128,
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).Dial,
+	},
+}
 
-var (
-	etcdPeers = os.Getenv("ETCD_PEERS")
-	domains   = os.Getenv("DOMAINS")
-)
+func main() {
+	app := cli.App("elb dns registrator", "Registers elb cname to *-up.ft.com cnames in dyn using konstructor")
+
+	domains := app.String(cli.StringOpt{
+		Name:   "domains",
+		Desc:   "comma separated *-up domains",
+		EnvVar: "DOMAINS",
+	})
+	konstructorBaseURL := app.String(cli.StringOpt{
+		Name:   "konstructor-base-url",
+		Desc:   "konstructor base url: https://dns-api.in.ft.com/v2",
+		EnvVar: "KONSTRUCTOR_BASE_URL",
+		Value:  "https://dns-api.in.ft.com/v2",
+	})
+	konstructorAPIKey := app.String(cli.StringOpt{
+		Name:   "konstructor-api-key",
+		Desc:   "konstructor api key",
+		EnvVar: "KONSTRUCTOR_API_KEY",
+	})
+	elbName := app.String(cli.StringOpt{
+		Name:   "elb-name",
+		Desc:   "elb cname",
+		EnvVar: "ELB_NAME",
+	})
+	awsRegion := app.String(cli.StringOpt{
+		Name:   "aws-region",
+		Desc:   "aws region",
+		EnvVar: "AWS_REGION",
+	})
+	awsAccessKeyID := app.String(cli.StringOpt{
+		Name:   "aws_access_key_id",
+		Desc:   "aws access key id",
+		EnvVar: "AWS_ACCESS_KEY_ID",
+	})
+	awsSecretAccessKey := app.String(cli.StringOpt{
+		Name:   "aws_secret_access_key",
+		Desc:   "aws secret access key",
+		EnvVar: "AWS_SECRET_ACCESS_KEY",
+	})
+
+	app.Action = func() {
+		c := &conf{
+			konsAPIKey:      *konstructorAPIKey,
+			konsDNSEndPoint: *konstructorBaseURL,
+			elbName:         *elbName,
+			awsAccessKey:    *awsAccessKeyID,
+			awsSecretKey:    *awsSecretAccessKey,
+			awsRegion:       *awsRegion,
+		}
+
+		elbCNAME := elbDNSName(c)
+		domainsToRegister := strings.Split(*domains, ",")
+
+		for _, domain := range domainsToRegister {
+			currentCNAME, err := getCurrentCNAME(c, domain)
+			if err != nil {
+				log.Fatal(err)
+			}
+			if currentCNAME == "" {
+				err = createDNS(c, elbCNAME, domain)
+				if err != nil {
+					log.Fatal(err)
+				}
+			} else {
+				err = updateDNS(c, currentCNAME, elbCNAME, domain)
+				if err != nil {
+					log.Fatal(err)
+				}
```

Comment:
Error handling for the `currentCNAME == ""` cases can be extracted so it's only done once.